### PR TITLE
Remove deprecated methods from JdbcTableHandle

### LINKED ISSUE #6797: Remove usages of deprecated JdbcTableHandle accessors
There are some existing deprecated methods and https://github.com/trinodb/trino/pull/6798 adds more deprecations.

### PR DESCRIPTION
**Pull request by tangjiangling**
## Related issues, pull requests, and links

Fixes #6797

## Documentation

(x) No documentation is needed.

## Release notes

(x) No release notes entries required.

### CODE REVIEW

**Review by ebyhr — COMMENTED** [review]
Please check CI failures.

**Comment by tangjiangling** [comment]
Squashed commits and fixed CI.

**Comment by hashhar** [comment]
How did you decide on when to use `asPlainTable` vs `getRequiredNamedRelation`?

File: plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClient.java [hunk 1]
```diff
@@ -473,8 +474,9 @@ private static Map<String, Integer> getArrayColumnDimensions(Connection connecti
                 "AND tbl.relname = ? " +
                 "AND attyp.typcategory = 'A' ";
         try (PreparedStatement statement = connection.prepareStatement(sql)) {
-            statement.setString(1, tableHandle.getSchemaName());
-            statement.setString(2, tableHandle.getTableName());
+            RemoteTableName remoteTableName = tableHandle.asPlainTable().getRemoteTableName();
```
Comment (hashhar) [inline plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClient.java:477]: The callers of this method ensure the handle is always a named relation so we can use `getRequiredNamedRelation`. Same in `getColumns` above.

(No changes requested for now - I'm confused myself about when to use one vs the other).

Comment (tangjiangling) [reply plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClient.java:477]: Good catch!

**Comment by tangjiangling** [comment]
> How did you decide on when to use asPlainTable vs getRequiredNamedRelation?

This is a really good question, and I'm probably not the best person to answer it (CC: @findepi).

But I can share my understanding:

- If a table is not synthetic, like in `JdbcClient#getTableHandle`,` JdbcClient#renameTable`,`JdbcClient#addColumn`,`JdbcClient#dropColumn` and other such method calls, it is obvious that "Table" will simply mean which table under which database, and will not contain other information (such as filter push-down, limit push-down, etc.), I would use `JdbcTableHandle#asPlainTable` (as in the naming description, it just describes which table under which database)
- If a table is synthetic (such as the optimizer's various push-downs: filter push-down, limit push-down, etc., which carry additional information than just a plain table), I would use `JdbcTableHandle#getRequiredNamedRelation`

To summarize, you need to know what state your table is in at the time of use (synthetic or non-synthetic) before you can choose whether to use `getRequiredNamedRelation` or `asPlainTable`.

File: plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/TestJdbcClient.java [hunk 2]
```diff
@@ -23,6 +23,8 @@
 import org.testng.annotations.BeforeClass;
 import org.testng.annotations.Test;
 
+import javax.ws.rs.HEAD;
```
Comment (hashhar) [inline plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/TestJdbcClient.java:26]: unused import

**Review by hashhar — APPROVED** [review]
LGTM % ci.

**Comment by hashhar** [comment]
I agree with your understanding of the usages @tangjiangling . However to me this seems like the API around JdbcTableHandle should be revisited at some point. The table handle today is basically a single class which can represent two different types of objects - actual tables as they exist in remote database or something represented by a `PreparedQuery` - maybe they should each be their own class and that way we can have the type-system guide us what kind of handle is expected where.

We'll discuss with Piotr once he's back and follow-up accordingly.

This PR looks good to me.

**Comment by tangjiangling** [comment]
(Fixed CI)

**Comment by tangjiangling** [comment]
CI hit #13362

**Comment by tangjiangling** [comment]
(Cherry-pick to resolve conflicts)

**Comment by tangjiangling** [comment]
(Cherry-pick to resolve conflicts again)

**Review by hashhar — APPROVED** [review]

**Comment by hashhar** [comment]
Thanks. I did check that all the deprecations have been there for at-least 2 years now (almost since the Trino rename).
